### PR TITLE
Production build

### PR DESCRIPTION
**Pull request by rkmarks**
**What kind of change does this PR introduce?**

- [ ] bugfix
- [ ] feature
- [ ] refactoring / style
- [x] build / chore
- [ ] documentation

**Did you add tests for your changes?**

- [ ] Yes, my code is well tested
- [x] Not relevant

**If relevant, did you update the documentation?**

- [ ] Yes, I've updated the documentation
- [x] Not relevant

**Summary**

#533 
**Does this PR introduce a breaking change?**

**Other information**

### CODE REVIEW

**Comment by okonet** [comment]
@rkmarks can you please write a summary of your changes and the goal of this PR so I can better understand the intent?

**Comment by rkmarks** [comment]
sorry @okonet i've been busy lately. the idea here is to use Uglify plugin on webpack to remove console lines and reduce our module only on production builds

**Comment by rkmarks** [comment]
@okonet i know you've been a bit busy but do you mind reviewing my comment above. any advice would be appreciated 😉

**Comment by okonet** [comment]
Does it affect the produced file size? If yes, lets update the `size-limit` config then? Otherwise LGTM!

File: package.json [hunk 1]
```diff
@@ -116,7 +117,9 @@
     "sinon": "^3.2.1",
     "size-limit": "^0.11.0",
     "style-loader": "^0.18.2",
-    "webpack": "^3.5.5"
+    "uglifyjs-webpack-plugin": "^1.0.1",
```
Comment (okonet) [inline package.json]: I can't see those dependencies required anywhere? Why are they needed?

File: package.json [hunk 2]
```diff
@@ -116,7 +117,9 @@
...
+    "webpack": "^3.5.5",
+    "webpack-merge": "^4.1.1"
```
Comment (okonet) [inline package.json]: And this.

**Comment by rkmarks** [comment]
@okonet can you please re-review when you have a chance. Thank you!

**Comment by okonet** [comment]
Can you please rebase it on top of master, squash unnecessary commits (esp. with merge conflicts inside) and rename them. Please remove the `small` etc from them. Thanks!

**Comment by rkmarks** [comment]
tried to rebase and squash but doesn't appear to work properly, i've destroyed my repo and will try again.

**Pull request closed without merging**